tests/load/pipeline/test_refresh_modes.py

```diff
@@ -0,0 +1,484 @@
+    with pipeline.sql_client() as client:
+        with client.execute_query("SELECT * FROM source_2_data_1 ORDER BY product") as cursor:
+            assert_only_table_columns(cursor, ["product"])
+            result = column_values(cursor, "product")
+            assert result == ["orange", "pear"]
+
+    # Second table from source 2 is gone
+    with pytest.raises(DatabaseUndefinedRelation):
+        with pipeline.sql_client() as client:
+            result = client.execute_sql("SELECT * FROM source_2_data_2")
+
+
+@pytest.mark.parametrize(
+    "destination_config",
+    destinations_configs(default_sql_configs=True, subset=["duckdb"]),
+    ids=lambda x: x.name,
+)
+def test_refresh_argument_to_run(destination_config: DestinationTestConfiguration):
+    pipeline = destination_config.setup_pipeline("refresh_full_test")
+
+    info = pipeline.run(refresh_source(first_run=True))
+    assert_load_info(info)
+
+    info = pipeline.run(
+        refresh_source(first_run=False).with_resources("some_data_3"),
+        refresh="drop_sources",
+    )
+    assert_load_info(info)
+
+    # Check local schema to confirm refresh was at all applied
+    tables = set(t["name"] for t in pipeline.default_schema.data_tables())
+    assert tables == {"some_data_3"}
+
+    # Run again without refresh to confirm refresh option doesn't persist on pipeline
+    info = pipeline.run(refresh_source(first_run=False).with_resources("some_data_2"))
+    assert_load_info(info)
+
+    # Nothing is dropped
+    tables = set(t["name"] for t in pipeline.default_schema.data_tables())
+    assert tables == {"some_data_2", "some_data_3"}
+
+
+@pytest.mark.parametrize(
+    "destination_config",
+    destinations_configs(default_sql_configs=True, subset=["duckdb"]),
+    ids=lambda x: x.name,
+)
+def test_refresh_argument_to_extract(destination_config: DestinationTestConfiguration):
+    pipeline = destination_config.setup_pipeline("refresh_full_test")
+
+    info = pipeline.run(refresh_source(first_run=True))
+    assert_load_info(info)
+
+    pipeline.extract(
+        refresh_source(first_run=False).with_resources("some_data_3"),
+        refresh="drop_sources",
+    )
+
+    tables = set(t["name"] for t in pipeline.default_schema.data_tables(include_incomplete=True))
+    # All other data tables removed
+    assert tables == {"some_data_3", "some_data_4"}
+
+    # Run again without refresh to confirm refresh option doesn't persist on pipeline
+    pipeline.extract(refresh_source(first_run=False).with_resources("some_data_2"))
+
+    tables = set(t["name"] for t in pipeline.default_schema.data_tables(include_incomplete=True))
+    assert tables == {"some_data_2", "some_data_3", "some_data_4"}
+
+
+@pytest.mark.parametrize(
+    "destination_config", destinations_configs(local_filesystem_configs=True), ids=lambda x: x.name
+)
+def test_refresh_drop_sources_local_filesystem(destination_config: DestinationTestConfiguration):
```

Comment:
imho this does not need to be tested in a special test for local filesystem. we have a bunch of utils where you can query the items for each table regardless of wether the destination is a filesystem or a sql db. or is there something here that will not work this way?

Reply:
Generally speaking it would be cool to have some kind of unified interface to query all items from a table that needs to be implemented for all destinations so we do not need these extra tests for filesystem that are still around in a couple of places. But that is for another PR.

Reply:
I managed to do this mostly with the `load_tables_to_dicts` utils. All tests run on filesystem now!